package/MDAnalysis/analysis/rms.py

```diff
@@ -488,6 +495,15 @@ def __init__(self, atomgroup, reference=None, select='all',
                     "N_ref={3}, N_traj={4}".format(
                         igroup, sel['reference'], sel['mobile'],
                         len(atoms['reference']), len(atoms['mobile'])))
+        # check weights
+        if not (iterable(self.weights) or self.weights in ("mass", None)):
+            raise TypeError("weights must be {'mass', None} or an iterable of the "
+                            "same size as the atomgroup.")
+        elif iterable(self.weights) and \
+             len(self.weights) != self.mobile_atoms.n_atoms:
+            raise TypeError("weights (length {0}) must be of same length as "
+                            "the mobile selection ({1})".format(
+                                len(self.weights), self.mobile_atoms.n_atoms))
```

Comment:
P.S.: @kain88-de feel free to review my PRs with a "request changes" instead of suggestive comments – I don't mind. If I feel it's too much or not appropriate I will argue but I think this is how it is supposed to work.